Add Websocket "config entries" methods

mj23000 opened the pull request:
Add most of the Websocket methods available in the [config_entries.py](https://github.com/home-assistant/core/blob/dev/homeassistant/components/config/config_entries.py) file.

Missing from this PR so far:
- ~Add testing for the newly added methods~
- ~Add `config_entries/subscribe` method~
- Ensure that `get_entry_subentries` and `delete_entry_subentry` work as intended

This PR also tweaks how the `from_json()` methods are used, so some feedback would be appreciated.

Kind regards

mj23000 committed:
Add from_json to BaseModel to reduce duplicated code

mj23000 committed:
Add from_json to BaseModel to reduce duplicated code

mj23000 committed:
Add method to get non-user flows in progress

mj23000 committed:
Add method to get non-user flows in progress

mj23000 committed:
Add disable/enable config entry methods

mj23000 committed:
Add disable/enable config entry methods

mj23000 committed:
Fix Websocket Error model by not requiring translation keys

mj23000 committed:
Add ignore flow method

mj23000 committed:
Add ignore flow method

mj23000 committed:
Add method to get a filtered list of config entries

mj23000 committed:
Add method to get a filtered list of config entries

mj23000 committed:
Add get entry subentry method

mj23000 committed:
Add get entry subentry method

mj23000 committed:
Add delete subentry method

mj23000 committed:
Add delete subentry method

mj23000 committed:
Fix typing

mj23000 committed:
Fix typing

GrandMoff100 commented on homeassistant_api/models/domains.py:
```diff
@@ -55,7 +56,14 @@ def __init__(
+    @override
+    def from_json(cls, json: Union[dict[str, JSONType], Any, None], **kwargs) -> Self:
+        raise ValueError(
+            f"`{cls.__name__}` does not support `from_json()`. Use `from_json_with_client()`"
```
This is a nifty idea to add `from_json_with_client` making the dependence on a client more explicit. I like it.

GrandMoff100 commented on homeassistant_api/websocket.py:
```diff
@@ -194,6 +204,98 @@ def get_domain(self, domain: str) -> Domain:
+            )
+        )
+
+    # TODO: config_entries/subscribe
```
Still seeing some TODOs left on this PR so I'm also going to convert this one to a draft. You can mark it ready for review when you're finished with the TODOs and adding tests. Let me know if you want any more specific guidance. I'm sorry that its taken me so long to get back to you on this, I've been trying to get ahead on my graduate coursework. Thank you so much again!

mj23000 committed:
Add Websocket endpoints to docstrings

mj23000 committed:
Add Websocket endpoints to docstrings

mj23000 committed:
Add config_entries subscribe method

mj23000 committed:
Add config_entries subscribe method

mj23000 committed:
Add testing

mj23000 committed:
Add testing

mj23000 commented:
I've added the missing method and testing for the new code.

GrandMoff100 committed:
Merge branch 'dev' into add_websocket_config_entries

adamlogan73 committed:
add missing testing.
Some updates to dependencies and conftest were also required.
Updated ruff version did formatting slightly different in a few files.

adamlogan73 committed:
Merge remote-tracking branch 'mj23000/add_websocket_config_entries' into fork/mj23000/add_websocket_config_entries

adamlogan73 committed:
update ruff commands in github cicd pipeline

GrandMoff100 committed:
Update Python build_from version  and fix websocket client fixture type

adamlogan73 committed:
Fix test coverage for logbook and entity history tests

Use sun.red_sun entity for logbook tests (sun.sun has no logbook entries).
Replace for/else raise pattern with direct assertions in history tests.
